prov/usnic/src/usdf_endpoint.c

```diff
@@ -76,3 +77,54 @@ usdf_endpoint_open(struct fid_domain *domain, struct fi_info *info,
 		return -FI_ENODEV;
 	}
 }
+
+int usdf_ep_getopt_connected(fid_t fid, int level, int optname, void *optval,
+		size_t *optlen)
+{
+	size_t *cm_size;
+	size_t dest_size;
+
+	USDF_TRACE_SYS(EP_CTRL, "\n");
+
+	if (!optval || !optlen)
+		return -FI_EINVAL;
+
+	if (level != FI_OPT_ENDPOINT)
+		return -FI_ENOPROTOOPT;
+
+	switch (optname) {
+	case FI_OPT_CM_DATA_SIZE:
+		dest_size = *optlen;
+		*optlen = sizeof(*cm_size);
+
+		if (dest_size < sizeof(*cm_size))
+			return -FI_ETOOSMALL;
+
+		cm_size = optval;
+		*cm_size = USDF_MAX_CONN_DATA;
+		break;
+	case FI_OPT_ENDPOINT:
```

Comment:
`FI_OPT_ENDPOINT` is a level constant, not an optname constant.  Please delete this `case`.

It's too late for this without breaking ABI, but in an ideal world I think we'd have separated the value space a bit so that the error checking could be the more user friendly (`FI_OPT_MIN_MULTI_RECV` and `FI_OPT_ENDPOINT` have the same value but different conceptual roles).  Another tweak that would have been good would have been changing the name to `FI_OPT_LVL_ENDPOINT' or something that indicates the constant is for "levels".  Oh well, it's a minor thing.